### PR TITLE
Operator-related fix and openshift v4 support

### PR DESCRIPTION
**Pull request by davidfestal**
### What does this PR do?

This PR provides fixes to the `che-keycloak` image in order to allow the community Che operator
to successfully install a Che server integrated with the Openshift v4 OAuth authentication.

More precisely this PR:
- Restores the initial user that is set in the base Keycloak image (1000 => jboss user), instead of keeping the ROOT user as the docker image user, which brings problems in both Kubernetes or Openshift contexts in the Che operator use-cases.
- Fixes an incompatibility with the Che operator introduced by PR https://github.com/eclipse/che/pull/13429 : the reason is that the last Keycloak image has moved its entry-point. Since the Che operator overrides the entry-point and then delegates to the original one, Che operator was broken by this change. In order to provide compatibility of the Che operator with as many Che versions as possible, this PR add a link between the new and the old entry-point paths, so that the Che operator will be able to work with both.    
- Deploys the new Openshift-v4 identity provider implemented in the context of issues https://issues.jboss.org/browse/CRW-202 and https://issues.jboss.org/browse/KEYCLOAK-10169.
When this new provider will be integrated into the Keycloak product itself in a future Keycloak release, we'll be able to remove this last change.  

### What issues does this PR fix or reference?

This PR is involved in fixing issue: https://github.com/redhat-developer/rh-che/issues/1454,
and is also a followup of PR https://github.com/eclipse/che/pull/13429/files which had introduced a breaking change for the community Che operator.

### CODE REVIEW

File: dockerfiles/keycloak/Dockerfile [hunk 1]
```diff
@@ -10,7 +10,14 @@ FROM jboss/keycloak:6.0.1
 ADD che /opt/jboss/keycloak/themes/che
 ADD . /scripts/
 ADD cli /scripts/cli
+RUN ln -s /opt/jboss/tools/docker-entrypoint.sh
```
Comment (benoitf) [inline dockerfiles/keycloak/Dockerfile]: Hi, can we merge all run instructions into a single one ?

File: dockerfiles/keycloak/Dockerfile [hunk 2]
```diff
@@ -10,7 +10,14 @@ FROM jboss/keycloak:6.0.1
...
+RUN curl --location https://github.com/davidfestal/KEYCLOAK-10169-OpenShift4-User-Provider/releases/download/6.0.1-openshift-v4-provider/openshift4-extension-6.0.1.jar -o /opt/jboss/keycloak/standalone/deployments/openshift4-extension-6.0.1.jar
```
Comment (benoitf) [inline dockerfiles/keycloak/Dockerfile]: we don't have jar on something that is not belonging to a given user (davidfestal ? )

Comment (benoitf) [reply dockerfiles/keycloak/Dockerfile]: maybe use a multi-staged build ?

Comment (nickboldt) [reply dockerfiles/keycloak/Dockerfile]: eventually the fix will be in Keycloak upstream (and RH-SSO downstream) but for now it's a custom jar from https://github.com/slaskawi/KEYCLOAK-10169-OpenShift4-User-Provider (or David's fork). It's not yet in the Keycloak/SSO codebase. See KEYCLOAK-10169 for discussion/details.

Comment (l0rd) [reply dockerfiles/keycloak/Dockerfile]: +1 to @benoitf suggestion. And consider using `curl -sSL` flags to follow redirections and avoid having the % for the download in the logs.

**Comment by davidfestal** [comment]
@benoitf @l0rd I fixed your comments. Is it better like that ?

**Review by benoitf — APPROVED** [review]

**Review by l0rd — APPROVED** [review]

**Review by l0rd — APPROVED** [review]

**Review by l0rd — APPROVED** [review]

**Pull request merged**